src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -60,21 +61,18 @@ def run_escalating_forecast() -> None:
 )
 @retry
 def generate_forecasts_for_projects(project_ids: List[int]) -> None:
-    query_until_escalating_groups = [
```

Comment:
You can change the `[` to `(` and it'll be a generator instead of a list, so you'll be able to iterate through it lazily

Reply:
This is what it was before for reference:

`for until_escalating_groups in chunked(
        RangeQuerySetWrapper(
            Group.objects.filter(
                status=GroupStatus.IGNORED,
                substatus=GroupSubStatus.UNTIL_ESCALATING,
                project_id__in=project_ids,
                last_seen__gte=datetime.now() - timedelta(days=7),
            ),
            step=ITERATOR_CHUNK,
        ),
        ITERATOR_CHUNK,
    ):
        generate_and_save_forecasts(groups=until_escalating_groups)`
        
`for until_escalating_groups in chunked` had RangeQuerySetWrapper directly inside. 
Correct me if I'm wrong, but in the code I have below, it was returning groups instead